### DISCUSSION TITLE
Inserimento CIG

### DISCUSSION THREAD

**Discussion by teo256** [fattureincloud/api#111] [Q&A] [answered]
Buongiorno a tutti,

usando il seguente codice C#:

```
 IssuedDocument invoice = new IssuedDocument(
                    type: IssuedDocumentType.Invoice,
                    entity: entity,
.....
eInvoice: true,
                    eiData: new IssuedDocumentEiData(
                       cig: valoreCIG,
                       cup: valoreCUP,
                       paymentMethod: this.fE_naturaTextBox.Text
                    ),
```

non viene inserito il CIG sulla fattura.

Da console sembra tutto ok:
  EiData: class IssuedDocumentEiData {
  VatKind: 
  OriginalDocumentType: 
  OdNumber: 
  OdDate: 
  Cig: ZA**********
  Cup: 
  PaymentMethod: MP05

Cosa sbaglio?

**Comment by emajo** [comment, marked as answer]
Ciao @teo256 per vedere il campo `cig` in fattura devono essere valorizzati anche i campi `original_document_type`, `od_number` e `od_date`.

Rimango a disposizione per ulteriori chiarimenti,
Emanuele

**Reply by teo256** [reply]
Risolto!
Grazie mille!!
